ApollosProject/apollos-apps
# Interactions performance improvements using Apollos Plugin

## DESCRIPTION

### What does this PR do, or why is it needed?

This PR updates the Interactions DataSource and all the methods that use it to fetch all interactions for a set of content items at once, using our ApollosPlugin (with no worries about node limits!) 


### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

1. Pull this PR, and the template repo after the Up Next PR is mergeable. 
2. Visit a content series content item. 
3. Looking in the console, you shouldn't see any repeated queries for interactions. 
 - Instead, look for something like this: 
```
 /Apollos/GetInteractionsByForeignKeys?keys=WeekendContentItem%3Ab22198037ab47338dbddca91eeffd911%2CWeekendContentItem%3A91d010bc55c718efb6de632ba1558eca%2CWeekendContentItem%3A836a837a1aec93fc163d45315340ff0c%2CWeekendContentItem%3A07fe9a897c8cb79276f906c8718ffe2a%2CWeekendContentItem%3Accb0cbeeb85d39a647e22440227f4733%2CWeekendContentItem%3Aea0a940c3ff6453e10508c3111dad1bb%2CWeekendContentItem%3A00bc2153acaeb246a790703f5ec5614f%2CWeekendContentItem%3Aacb77fcf685b3a24f111e187f485453b&%24filter=((Operation eq 'COMPLETE')) and (PersonAliasId eq 284): 1
```

### What automated tests did you write?

New tests to cover new functionality. 

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
